x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/AnthropicCompletionRequestManager.java

```diff
@@ -43,13 +41,13 @@ private AnthropicCompletionRequestManager(AnthropicChatCompletionModel model, Th
 
     @Override
     public void execute(
-        @Nullable String query,
-        List<String> input,
+        InferenceInputs inferenceInputs,
         RequestSender requestSender,
         Supplier<Boolean> hasRequestCompletedFunction,
         ActionListener<InferenceServiceResults> listener
     ) {
-        AnthropicChatCompletionRequest request = new AnthropicChatCompletionRequest(input, model);
+        var docsInput = DocumentsOnlyInput.of(inferenceInputs);
+        AnthropicChatCompletionRequest request = new AnthropicChatCompletionRequest(docsInput.getInputs(), model);
```

Comment:
I found this code slightly obscure because I have to go look at the DocumentsOnlyInput class to find out what's happening here with the types. What do you think of this instead:
```suggestion
        List<String> docsInput = DocumentsOnlyInput.of(inferenceInputs).getInputs();
        AnthropicChatCompletionRequest request = new AnthropicChatCompletionRequest(docsInput, model);
```

Reply:
this also applies to all of the other files with this change, unfortunately